src/selection-recommendation/recommend.py

```diff
@@ -0,0 +1,130 @@
+import sys
+sys.path.append("../spider/util")
+import itertools
+import pymysql
+import Database
+import time_conflicts_check
+
+DB_NAME = "uoftcourses"
+DB_PATH = "../../database.info"
+
+
+def get_course_data(course_code, campus):
+    """
+    course_code: a string e.g. "CSC148"
+    campus: a string - either "St. George", "Scarborough", or "Mississauga"
+    Returns a list of tuples, with each tuple containing the data of a single section of the specified course
+    """
+    connection = Database.get_connection(DB_PATH, DB_NAME)
+    cursor = connection.cursor()
+    sql = "SELECT * FROM Course WHERE cID like %s AND campus=%s"
+    cursor.execute(sql, ("%" + course_code + "%", campus))
+
+    return(list(cursor.fetchall()))
+
+
+def get_all_possible_course_times(course_code, campus):
+    """
+    course_code: a string e.g. "CSC148"
+    campus: a string - either "St. George", "Scarborough", or "Mississauga"
+    Returns a tuple:
+        - the first element is a list of strings containing all possible times (i.e. no conflicts) of combinations of lecture/tutorial/lab sections.
+          For the format of the strings, see the process_times() function in the time_conflicts_check module
+        - the second element is a list of tuples of strings representing the respective lecture, tutorial and lab sections
+
+    Note that the indices of the two lists correspond. For example, the first element of the first list has times that correspond
+    to the class sections in the first element of the second list.
+    """
+    all_course_data = get_course_data(course_code, campus)
+
+    lecture_times = []
+    lecture_sections = []
+    tutorial_times = []
+    tutorial_sections = []
+    lab_times = []
+    lab_sections = []
+
+    for course_data in all_course_data:
+        if course_data[15] and course_data[14]:
+            if int(course_data[15]) >= int(course_data[14]):  # check if section is full
```

Comment:
Well, the enrollment num data is usually not up to date, neither data we collected in our database. Also, enrollment num may be null. So it's better to delete this line